lib_neutral_prompt/cfg_denoiser_hijack.py

```diff
@@ -212,27 +212,111 @@ def filter_abs_top_k(vector: torch.Tensor, k_ratio: float) -> torch.Tensor:
     return vector * (torch.abs(vector) >= top_k).to(vector.dtype)
 
 
-sd_samplers_hijacker = hijacker.ModuleHijacker.install_or_get(
-    module=sd_samplers,
-    hijacker_attribute='__neutral_prompt_hijacker',
-    on_uninstall=script_callbacks.on_script_unloaded,
-)
+try:
+    from ldm_patched.modules import samplers
+    from modules_forge import forge_sampler
+    forge = True
+except ImportError:
+    forge = False
+
+
+if forge:
+    forge_sampler_hijacker = hijacker.ModuleHijacker.install_or_get(
+        module=forge_sampler,
+        hijacker_attribute='__forge_sample_hijacker',
+        on_uninstall=script_callbacks.on_script_unloaded,
+    )
+
+    ldm_patched_modules_hijacker = hijacker.ModuleHijacker.install_or_get(
+        module=samplers,
+        hijacker_attribute='__ldm_patched_modules_hijacker',
+        on_uninstall=script_callbacks.on_script_unloaded,
+    )
 
 
-@sd_samplers_hijacker.hijack('create_sampler')
-def create_sampler_hijack(name: str, model, original_function):
-    sampler = original_function(name, model)
-    if not hasattr(sampler, 'model_wrap_cfg') or not hasattr(sampler.model_wrap_cfg, 'combine_denoised'):
-        if global_state.is_enabled:
-            warn_unsupported_sampler()
+    @forge_sampler_hijacker.hijack('forge_sample')
+    def forge_sample(self, denoiser_params, cond_scale, cond_composition, original_function):
+        if not global_state.is_enabled:
+            return original_function(self, denoiser_params, cond_scale, cond_composition)
 
-        return sampler
+        self.inner_model.inner_model.forge_objects.unet.model_options['cond_composition'] = cond_composition
```

Comment:
TODO: do this with the patcher instead, drop hijacking of forge_sample completely

Reply:
I guess we won't drop hijacking for now, I'll resolve this.